opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -453,23 +453,17 @@ def resource(self) -> Resource:
     @deprecated(
         version="1.11.1", reason="You should use instrumentation_scope"
     )
-    def instrumentation_info(self) -> InstrumentationInfo:
+    def instrumentation_info(self) -> Optional[InstrumentationInfo]:
         return self._instrumentation_info
 
     @property
-    def instrumentation_scope(self) -> InstrumentationScope:
+    def instrumentation_scope(self) -> Optional[InstrumentationScope]:
         return self._instrumentation_scope
 
-    def to_json(self, indent=4):
+    def to_json(self, indent: int = 4):
         parent_id = None
         if self.parent is not None:
-            if isinstance(self.parent, Span):
-                ctx = self.parent.context
-                parent_id = f"0x{trace_api.format_span_id(ctx.span_id)}"
-            elif isinstance(self.parent, SpanContext):
```

Comment:
Any idea why this check was needed before?

Reply:
The constructor says it always has to be a `SpanContext`. I found a test that was passing in a `Span`, so I changed the test and assumed the constructor is right. I'm happy to do the other way around and assume it can be either but that adds more complexity and IMO it seems like the parent's context is the right thing to pass in, not the parent itself.

Reply:
https://github.com/open-telemetry/opentelemetry-python/pull/3528/files#r1394750050

Reply:
for what it's worth the blame seems to indicate this has been like this since `ReadableSpan` was first written